src/main/java/seedu/address/logic/commands/DeleteTagCommand.java

```diff
@@ -57,85 +57,101 @@ public CommandResult execute(Model model) throws CommandException {
-                    showSuccessMsg = true;
-                }
-            }
-
-            successMsg.append(MESSAGE_SUCCESS_FORCE);
-            String successMessage = successMsg.toString();
-
-            if (showErrorMsg) {
-                errorMsg.append(MESSAGE_NONEXISTENT);
-                String errorMessage = errorMsg.toString();
-                if (showSuccessMsg) {
-                    return new CommandResult(successMessage + " " + errorMessage);
-                }
-                return new CommandResult(errorMessage);
-            }
-            return new CommandResult(successMessage);
-
+            showSuccessMsg = deleteTagsAndSuccess(model, successMsg, MESSAGE_SUCCESS_FORCE);
         } else {
-            // Check if any tags are in use
+            // If not force delete, check for tags in use and handle accordingly
             Set<Tag> tagsInUse = model.getTagsInUse();
             Set<Tag> matchingTags = tagsInUse.stream().filter(tag -> tags.contains(tag)).collect(Collectors.toSet());
 
             if (!matchingTags.isEmpty()) {
                 return new CommandResult(String.format(MESSAGE_TAGS_IN_USE, matchingTags));
             }
 
-            for (Tag tag : tags) {
-                if (!model.hasTag(tag)) {
-                    errorMsg.append(tag.toString() + " ");
-                    showErrorMsg = true;
-                }
+            // Perform standard deletion and build success message
+            showSuccessMsg = deleteTagsAndSuccess(model, successMsg, MESSAGE_SUCCESS);
+        }
+        return constructFinalMessage(showErrorMsg, errorMsg, showSuccessMsg, successMsg);
+    }
+
+    /**
+     * Helper to add non-existent tags to the error message
+     *
+     * @param model
+     * @param errorMsg
+     * @return
+     */
+    private boolean addNonExistentTagsToError(Model model, StringBuilder errorMsg) {
+        boolean hasErrors = false;
+        for (Tag tag : tags) {
+            if (!model.hasTag(tag)) {
+                errorMsg.append(tag.toString()).append(" ");
+                hasErrors = true;
             }
+        }
+        if (hasErrors) {
+            errorMsg.append(MESSAGE_NONEXISTENT);
+        }
+        return hasErrors;
+    }
 
-            // Standard delete logic
-            for (Tag tag : tags) {
-                if (model.hasTag(tag)) {
-                    model.deleteTags(List.of(tag)); // Just delete from the tag list
-                    showSuccessMsg = true;
-                }
 
+    /**
+     * Helper method to delete tags and add to success message
+     *
+     * @param model
+     * @param successMsg
+     * @param successMessage
+     * @return
+     */
+    private boolean deleteTagsAndSuccess(Model model, StringBuilder successMsg, String successMessage) {
+        boolean hasSuccess = false;
+        for (Tag tag : tags) {
+            if (model.hasTag(tag)) {
+                model.deleteTags(List.of(tag)); // Remove from model tags
```

Comment:
can the deleteTags method be used directly on the list of tags instead of having the loop? also, the deleteTags method currently returns true if everything was successfully deleted (ie. everything exists) and false if it was not (ie. something does not exist) and no exception is thrown either way, so the additional check for hasTag() before every tag might not be necessary